src/test/CMakeLists.txt

```diff
@@ -1,3 +1,10 @@
+# -force_load/--whole-archive are necessary to include all symbols so that dynamically loaded plugins can use them
```

Comment:
@Bensk1 Conditionally defining HYRISE_LIBRARY_WITH_ALL_SYMBOLS in each CMakeLists.txt (the same cmake code as added in this PR appears in another CMakeLists.txt as well) in which libhyrise is linked to something is not a sustainable solution. Is there a way to modify the `hyrise` target in a way that linking to it automatically incurs these extra flags?

Reply:
I would not have any idea how the modification of the `hyrise` target should work. Could define `HYRISE_LIBRARY_WITH_ALL_SYMBOLS` in the topmost CMakeLists.txt and have it available in the other ones?

Reply:
I will have a look at it tomorrow.